cfn-resources/federated-query-limit/cmd/resource/resource.go

```diff
@@ -0,0 +1,334 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progress_events "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"github.com/spf13/cast"
+	atlasSDK "go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+var CreateOrUpdateRequiredFields = []string{constants.ProjectID, constants.TenantName, constants.LimitName, constants.Value}
+var ReadRequiredFields = []string{constants.ProjectID, constants.TenantName, constants.LimitName}
+var DeleteRequiredFields = []string{constants.ProjectID, constants.TenantName, constants.LimitName}
+var ListRequiredFields = []string{constants.ProjectID, constants.TenantName}
+
+const (
+	AlreadyExists = "already exists"
+	DoesntExists  = "does not exist"
+	CREATE        = "CREATE"
+	READ          = "READ"
+	UPDATE        = "UPDATE"
+	DELETE        = "DELETE"
+	LIST          = "LIST"
+)
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-federated-query-limit")
+}
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(CreateOrUpdateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlas, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	_, response, err := getFederatedQueryLimit(atlas, currentModel)
+	defer closeResponse(response)
+	if err == nil {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          AlreadyExists,
+			HandlerErrorCode: cloudformation.HandlerErrorCodeAlreadyExists}, nil
```

Comment:
here we are assuming that whenever an error comes it will be a not found

Reply:
added 404 error check as well now